cmd/snap/cmd_interface.go

```diff
@@ -179,7 +180,7 @@ func (x *cmdInterface) showAttrs(w io.Writer, attrs map[string]interface{}, inde
 	for _, name := range names {
 		value := attrs[name]
 		switch value.(type) {
-		case string, int, bool:
+		case string, bool, json.Number:
```

Comment:
Hmm, so we never ever see int? Just always json.Number?

Reply:
I can write more if needed, but yeah: we use the json decoder with UseNumber() which produces json.Number output. But even if we just would use normal decoding this code is not correct, because the json decoder will produce float64 for numbers and not ints.

Reply:
yes, otoh this is a quick fix, I'm a bit skeptical that having Numbers around in some parts of client results is sane long term

Reply:
If the preferred way to fix this is to rework the client to not have json.Numbers (maybe except for the "client.Conf" parts) then that is fine with me, I will close the PR in this case.

Reply:
I don't know, we don't have a strong use case yet to do the more complicated natural thing, which I think would be to parse  to Number and then normalize to either int64 or float64 like we do when parsing snap.yaml

Reply:
When config will have schemas we will also have to reconsider the whole thing.

Reply:
FWTW, json.Numbers all around were introduced with https://github.com/snapcore/snapd/pull/3569 to fix specific class of issues reported back then (forum link in the PR).